### PR TITLE
Change the way Icon is wrapped

### PR DESCRIPTION
**Pull request by jordankoschei-okta**
`Icon` was one of the earlier components we wrapped, so there are a bunch of stylistic things that needed to change. This update changes the following:
- The component no longer inherits props from `MUISvgIcon`. It's self-contained.
- I restricted the colors that could be assigned to an icon. Most of the time it'll still be `inherit` (which it is in all current usages, afaik) but there are options for `success`, `warning`, `error`, `info`, and `primary` colors too. (I disabled `action` and `secondary`, as those aren't used in our color scheme)
- I switched `titleAccess` with `ariaLabelledby` and `label` for clarity. `label` is required, and `ariaLabelledby` should be used if there's some text adjacent to the icon that describes its purpose.
- I rolled in some `size` changes from another PR

### CODE REVIEW

**Comment by jordankoschei-okta** [comment]
Remove `color` from Icon props. It should only ever be `inherit`

File: packages/odyssey-react-mui/src/Icon.tsx [hunk 1]
```diff
@@ -28,12 +43,19 @@ export interface IconProps extends SvgIconNoChildrenProps {
  * icon component imports where possible to keep your bundle size smaller.
  */
 export const Icon = forwardRef<SVGSVGElement, IconProps>(
```
Comment (jordankoschei-okta) [inline packages/odyssey-react-mui/src/Icon.tsx:45]: Remove forwardRef

File: packages/odyssey-storybook/src/components/odyssey-mui/Icon/Icon.stories.tsx [hunk 2]
```diff
@@ -10,43 +10,65 @@
...
 import { MuiThemeDecorator } from "../../../../.storybook/components/MuiThemeDecorator";
 
-export default {
+const storybookMeta: Meta<IconProps> = {
   title: `MUI Components/Icon`,
   component: Icon,
   decorators: [MuiThemeDecorator],
+  argTypes: {
+    name: {
+      defaultValue: "alert-triangle-filled",
```
Comment (jordankoschei-okta) [inline packages/odyssey-storybook/src/components/odyssey-mui/Icon/Icon.stories.tsx:35]: Move defaultValue to args

Comment (jordankoschei-okta) [reply packages/odyssey-storybook/src/components/odyssey-mui/Icon/Icon.stories.tsx:35]: And pull in `develop`!

**Review by KevinGhadyani-Okta — APPROVED** [review]
Approved, but some small changes required which we met about. You already put comments in for those.

**Pull request merged**